runtime/appruntime/apisdk/api/handler.go

```diff
@@ -691,15 +681,20 @@ func (d *Desc[Req, Resp]) externalCall(c CallContext, service config.Service, re
 				if err != nil {
 					return resp, errs.B().Code(errs.Internal).Msg("request failed: unable to read response").Err()
 				}
-				return resp, errs.B().Code(errs.Internal).Msg("request failed: " + string(bodyBytes)).Err()
+
+				if len(bodyBytes) == 0 {
+					return resp, errs.B().Code(errs.Internal).Msgf("request failed: status %d (%s)", httpResp.StatusCode, httpResp.Status).Err()
+				} else {
+					return resp, errs.B().Code(errs.Internal).Msgf("request failed: status %d: %s", httpResp.StatusCode, string(bodyBytes)).Err()
+				}
 			}
 		}
 
 		return d.DecodeExternalResp(httpResp, c.server.json)
 	})()
 
 	if respErr != nil {
-		c.server.rootLogger.Err(respErr).Msg("call failed")
+		c.server.rootLogger.Err(respErr).Str("calling", fmt.Sprintf("%s.%s", d.Service, d.Endpoint)).Str("url", reqURL).Msg("call failed")
```

Comment:
`calling` is a confusing name since it's also a present tense verb, but the call already failed. How about `target`?

```suggestion
		c.server.rootLogger.Err(respErr).Str("target", fmt.Sprintf("%s.%s", d.Service, d.Endpoint)).Str("url", reqURL).Msg("call failed")
```